### PR TITLE
Balder/inline frequently called methods

### PR DESCRIPTION
**Pull request by baldersheim**
@arnej27959 or @havardpe PR

### CODE REVIEW

File: eval/src/vespa/eval/tensor/sparse/sparse_tensor_address_builder.h [hunk 1]
```diff
@@ -19,16 +18,26 @@ namespace vespalib::tensor {
...
 private:
-    std::vector<char> _address;
+    vespalib::Array<char> _address;
 
+protected:
     void append(vespalib::stringref str) {
-        const char *cstr = str.c_str();
-        _address.insert(_address.end(), cstr, cstr + str.size() + 1);
+        for (size_t i(0); i < str.size() + 1; i++) {
+            _address.push_back_fast(str[i]);
```
Comment (havardpe) [inline eval/src/vespa/eval/tensor/sparse/sparse_tensor_address_builder.h]: I do not think accessing the 0-terminator like this is defined.

Comment (havardpe) [reply eval/src/vespa/eval/tensor/sparse/sparse_tensor_address_builder.h]: also, better would be to insert explicit 0 after the string, to also support appending substrings...

**Comment by havardpe** [comment]
ok

File: eval/src/vespa/eval/tensor/sparse/sparse_tensor_address_builder.h (hunk 1 above)
Comment (baldersheim) [reply eval/src/vespa/eval/tensor/sparse/sparse_tensor_address_builder.h]: I agree, just made a minimal change. I will fix it at next crossroads.

**Pull request merged**